x/gov/simulation/operations.go

```diff
@@ -0,0 +1,311 @@
+			return simulation.NoOpMsg(types.ModuleName), nil, err
+		}
+
+		msg := types.NewMsgDeposit(simAccount.Address, proposalID, deposit)
+
+		account := ak.GetAccount(ctx, simAccount.Address)
+		fees, err := helpers.RandomFees(r, ctx, account, deposit)
+		if err != nil {
+			return simulation.NoOpMsg(types.ModuleName), nil, err
+		}
+
+		tx := helpers.GenTx(
+			[]sdk.Msg{msg},
+			fees,
+			chainID,
+			[]uint64{account.GetAccountNumber()},
+			[]uint64{account.GetSequence()},
+			[]crypto.PrivKey{simAccount.PrivKey}...,
+		)
+
+		res := app.Deliver(tx)
+		if !res.IsOK() {
+			return simulation.NoOpMsg(types.ModuleName), nil, errors.New(res.Log)
+		}
+
+		return simulation.NewOperationMsg(msg, true, ""), nil, nil
+	}
+}
+
+// SimulateMsgVote generates a MsgVote with random values.
+func SimulateMsgVote(ak types.AccountKeeper, k keeper.Keeper) simulation.Operation {
+	return operationSimulateMsgVote(ak, k, simulation.Account{}, -1)
+}
+
+func operationSimulateMsgVote(ak types.AccountKeeper, k keeper.Keeper, simAccount simulation.Account, proposalIDInt int64) simulation.Operation {
+	return func(r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simulation.Account,
+		chainID string) (opMsg simulation.OperationMsg, fOps []simulation.FutureOperation, err error) {
+
+		if simAccount.Equals(simulation.Account{}) {
+			simAccount, _ = simulation.RandomAcc(r, accs)
+		}
+
+		var proposalID uint64
+
+		switch {
+		case proposalIDInt < 0:
+			var ok bool
+			proposalID, ok = randomProposalID(r, k, ctx, types.StatusVotingPeriod)
+			if !ok {
+				return simulation.NoOpMsg(types.ModuleName), nil, nil
+			}
+		default:
+			proposalID = uint64(proposalIDInt)
+		}
+
+		option := randomVotingOption(r)
+
+		msg := types.NewMsgVote(simAccount.Address, proposalID, option)
+
+		account := ak.GetAccount(ctx, simAccount.Address)
+		fees, err := helpers.RandomFees(r, ctx, account, nil)
+		if err != nil {
+			return simulation.NoOpMsg(types.ModuleName), nil, err
+		}
+
+		tx := helpers.GenTx(
+			[]sdk.Msg{msg},
+			fees,
+			chainID,
+			[]uint64{account.GetAccountNumber()},
+			[]uint64{account.GetSequence()},
+			[]crypto.PrivKey{simAccount.PrivKey}...,
+		)
+
+		res := app.Deliver(tx)
+		if !res.IsOK() {
+			return simulation.NoOpMsg(types.ModuleName), nil, errors.New(res.Log)
+		}
+
+		return simulation.NewOperationMsg(msg, true, ""), nil, nil
+	}
+}
+
+// Pick a random deposit
+func randomDeposit(r *rand.Rand, ctx sdk.Context, ak types.AccountKeeper, k keeper.Keeper, addr sdk.AccAddress,
+) (deposit sdk.Coins, skip bool, err error) {
+
+	account := ak.GetAccount(ctx, addr)
+	coins := account.SpendableCoins(ctx.BlockHeader().Time)
+	if coins.Empty() {
+		return nil, true, nil // skip
+	}
+
+	minDeposit := k.GetDepositParams(ctx).MinDeposit
+	denomIndex := r.Intn(len(minDeposit))
+	denom := minDeposit[denomIndex].Denom
+
+	depositCoins := coins.AmountOf(denom)
+	if depositCoins.IsZero() {
+		return nil, true, nil
+	}
+
+	var maxAmt sdk.Int
```

Comment:
You can avoid the switch by:

```suggestion
	maxAmt := depositCoins
```